### ISSUE TITLE
Goodness of Fit

### ISSUE THREAD

**Issue by bonanomi** [bonanomi/FiducialXSFWK#50]
Test feasibility of GoF (AD or KS). Probably this should do:

```
combine -M GoodnessOfFit --algo=KS SM_125_all_13TeV_xs_mass4l_bin_v3.root --setParameters r_smH_mass4l_0=1 -m 125.38 --freezeParameters MH -t 100 -s 102 --toysFrequentist

combine -M GoodnessOfFit --algo=KS SM_125_all_13TeV_xs_mass4l_bin_v3.root --setParameters r_smH_mass4l_0=1 -m 125.38 --freezeParameters MH -s 102
```

and than make the integral for `t > gof_data`.

**Comment by bonanomi** [comment]
Investigating for unbinned datasets with >1 POI. 

For the computation of the p-values one can use (e.g. here with Njets):

```
combine -M MultiDimFit --algo=singles SM_125_all_13TeV_xs_njets_pt30_eta4p7_bin_v3.root -m 125.38 --freezeParameters MH -n bestfit_nj --saveWorkspace --redefineSignalPOIs r_smH_NJ_0,r_smH_NJ_1,r_smH_NJ_2,r_smH_NJ_3,r_smH_NJ_4 --setParameterRanges MH=125.38,125.38:r_smH_NJ_0=0,5:r_smH_NJ_1=0,5:r_smH_NJ_2=0,5:r_smH_NJ_3=0,5:r_smH_NJ_4=0,5 --setParameters MH=125.38,K1Bin0=0.999349313569,K2Bin0=0.999130294507,K1Bin1=0.999351449988,K2Bin1=0.99913250755,K1Bin2=0.99934244611,K2Bin2=0.999131697983,K1Bin3=0.99934840293,K2Bin3=0.999132202016,K1Bin4=0.999363130186,K2Bin4=0.999117426042,r_smH_NJ_0=1,r_smH_NJ_1=1,r_smH_NJ_2=1,r_smH_NJ_3=1,r_smH_NJ_4=1 

combineTool.py higgsCombinebestfit_nj.MultiDimFit.mH125.38.root --snapshotName "MultiDimFit" -M MultiDimFit -m 125.38 --saveWorkspace -n DataSMCompat_nj --redefineSignalPOIs r_smH_NJ_0,r_smH_NJ_1,r_smH_NJ_2,r_smH_NJ_3,r_smH_NJ_4 --X-rtd MINIMIZER_freezeDisassociatedParams --algo fixed --fixedPointPOIs K1Bin0=0.999349313569,K2Bin0=0.999130294507,K1Bin1=0.999351449988,K2Bin1=0.99913250755,K1Bin2=0.99934244611,K2Bin2=0.999131697983,K1Bin3=0.99934840293,K2Bin3=0.999132202016,K1Bin4=0.999363130186,K2Bin4=0.999117426042,r_smH_NJ_0=1,r_smH_NJ_1=1,r_smH_NJ_2=1,r_smH_NJ_3=1,r_smH_NJ_4=1 --saveNLL --skipInitialFit

from scipy.stats import chi2

chi2pdf = chi2(6) # nr_bins (+ 1 if underflowExists)

deltaNLLSMCompat = 0.9922884 # limit->Scan("deltaNLL")

pvalSMCompat = 1 - chi2pdf.cdf(2*deltaNLLSMCompat)

pvalSMCompat
```

**Comment by bonanomi** [comment]
Closing this issue, as we agreed upon using compatibility test with SM as a metric to assess the GoF and the quality of the results. The command is linked above here.
